### PR TITLE
2.12.0-RC2

### PR DESCRIPTION
**Pull request by adriaanm**

### CODE REVIEW

File: news/_posts/2016-10-18-release-notes-2.12.0-RC2.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  - [#5398](https://github.com/scala/scala/pull/5398) [SD-225](https://github.com/scala/scala-dev/issues/225) Use a `lzycompute` method for module initialization. Address a performance regression;
+  - [#5417](https://github.com/scala/scala/pull/5417) [SD-233](https://github.com/scala/scala-dev/issues/233) `synchronized` blocks are JIT-friendly again;
+  - [#5433](https://github.com/scala/scala/pull/5433) Don't deprecate `Either.left` and `Either.right` yet;
+  - [#5392](https://github.com/scala/scala/pull/5392) [SI-9918](https://issues.scala-lang.org/browse/SI-9918) Don't crash on `object` in `trait` mixed into package object;
+  - [#5397](https://github.com/scala/scala/pull/5397) [SI-9920](https://issues.scala-lang.org/browse/SI-9920) Avoid linkage errors with captured local objects + self types;
+  - [#5430](https://github.com/scala/scala/pull/5430) Emit `object` in method like `lazy val`;
+  - [#5442](https://github.com/scala/scala/pull/5442) [SI-9943](https://issues.scala-lang.org/browse/SI-9943) `sealed` class does not yield SAM type;
+
+
+The most notable changes since M5 are:
```
Comment (SethTisue) [inline news/_posts/2016-10-18-release-notes-2.12.0-RC2.md]: not necessary to repeat this from the RC1 announcement, I think

File: news/_posts/2016-10-18-release-notes-2.12.0-RC2.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    of `def`, `val`, and `lazy val` handling, fixing assorted corner
+    cases and inconsistencies
+  - [#5311](https://github.com/scala/scala/pull/5311): Scala is now built with
+    sbt instead of Ant (affects only contributors, not users)
+
+
+In total, we merged [29 pull requests](https://github.com/scala/scala/pulls?q=is%3Apr+is%3Amerged+milestone%3A2.12.0-RC2).
+This milestone resolves [6 JIRA tickets](https://issues.scala-lang.org/issues/?jql=project%20%3D%20SI%20AND%20status%20%3D%20CLOSED%20AND%20resolution%20%3D%20Fixed%20AND%20fixVersion%20%3D%20%22Scala%202.12.0-RC2%22%20ORDER%20BY%20component%20ASC%2C%20priority%20DESC) and [9 scala-dev issues](https://github.com/scala/scala-dev/milestone/9?closed=1).
+
+As usual for Scala pre-releases, 2.12.0-RC2 is not guaranteed to be binary compatible with any other Scala version, including any 2.12 milestones.
```
Comment (SethTisue) [inline news/_posts/2016-10-18-release-notes-2.12.0-RC2.md]: milestones or release candidates

File: news/_posts/2016-10-18-release-notes-2.12.0-RC2.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+* If a closure is allocated and invoked within the same method, the closure invocation is replaced by an invocations of the corresponding lambda body method
+* Dead code elimination and a small number of cleanup optimizations
+* Box/unbox elimination [#4858](https://github.com/scala/scala/pull/4858)
+
+#### Either is now right-biased
+
+`Either` now supports operations like `map`, `flatMap`, `contains`,
+`toOption`, and so forth, which operate on the right-hand side.
+
+(`.left` and `.right` will be deprecated in favor of `.swap` in a later release.)
```
Comment (SethTisue) [inline news/_posts/2016-10-18-release-notes-2.12.0-RC2.md]: s/will be/may be/ — I think `.right` will definitely be deprecated but the situation with `.left` is less clear

File: news/_posts/2016-10-18-release-notes-2.12.0-RC2.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+Scala 2.12 is not and will not be binary compatible with the 2.11.x series.  This allows us to keep improving the Scala compiler and standard library.  We are working with the community to ensure that core projects in the Scala eco-system become available for 2.12.  Please refer to this growing [list of libraries and frameworks](https://github.com/scala/make-release-notes/blob/2.12.x/projects-2.12.md).
+
+The [Scala 2.11.1 release notes](http://scala-lang.org/news/2.11.1) explain in more detail on how binary compatibility works in Scala.  The same policies apply to 2.12 as well.
+
+
+## Contributors
+
+A big thank you to everyone who's helped improve Scala by reporting bugs, improving our documentation, spreading kindness in mailing lists and other public fora, and submitting and reviewing pull requests! You are all magnificent.
+
+According to `git shortlog -sn --no-merges v2.12.0-RC1..v2.12.0-RC2`, the following contributors helped to realize this milestone: Adriaan Moors, Jason Zaugg, Lukas Rytz, Seth Tisue, Stefan Zeiger, Antoine Gourlay, Raphael Jolly. Thank you!
```
Comment (SethTisue) [inline news/_posts/2016-10-18-release-notes-2.12.0-RC2.md]: s/milestone/release candidate/

**Comment by adriaanm** [comment]
Thanks. I've made your suggested changes.

**Pull request merged**